spirv_glsl.cpp

```diff
@@ -3894,14 +3913,20 @@ void CompilerGLSL::emit_instruction(const Instruction &instruction)
 		break;
 
 	// Textures
-	case OpImageSampleImplicitLod:
 	case OpImageSampleExplicitLod:
-	case OpImageSampleProjImplicitLod:
 	case OpImageSampleProjExplicitLod:
-	case OpImageSampleDrefImplicitLod:
 	case OpImageSampleDrefExplicitLod:
-	case OpImageSampleProjDrefImplicitLod:
 	case OpImageSampleProjDrefExplicitLod:
+	{
+		emit_texture_op(instruction);
+		if (is_legacy_es())
```

Comment:
I would move this test inside emit_texture_op. It is virtual and it doesn't make sense to run this for C++/Metal backends.